packages/itwin/geo-tools/src/GoogleAddressProvider.ts

```diff
@@ -0,0 +1,128 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { Cartographic } from "@itwin/core-common";
+import { IModelApp } from "@itwin/core-frontend";
+
+import type { MapCartoRectangle } from "@itwin/core-frontend";
+import type { AddressProvider, AddressRequest, AddressData, GeoCoder } from "./AddressProvider";
+
+
+export interface GoogleAddressData  extends AddressData {
+  placeId: string;
+}
+
+export class GoogleAddressProvider implements AddressProvider {
+  private _radius = 5000;
+  private _apiKey: string;
+  public readonly hasAddressIds = true;
+
+  constructor(locationBiasRadius?: number) {
+    if (locationBiasRadius !== undefined) {
+      this._radius = locationBiasRadius;
+    }
+
+    this._apiKey = "";
+    if (IModelApp.mapLayerFormatRegistry?.configOptions?.GoogleMaps) {
+      this._apiKey = IModelApp.mapLayerFormatRegistry?.configOptions?.GoogleMaps.value;
+    }
+  }
+
+  public supportsAddressLocation(): this is GeoCoder  {
+    return true;
+  }
+
+  protected async getAuthRequestHeader(): Promise<Record<string, string>> {
+    return { "X-Goog-Api-Key": this._apiKey};
+  }
+
+  protected async getPlacesBaseUrl(): Promise<string> {
+    return "https://places.googleapis.com/v1/places/";
```

Comment:
any attributions we need to show when using this?

Reply:
The attribution is the Google logo display on the base, hence the google base map most be used.